### PR TITLE
React SearchBox documentation

### PR DESCRIPTION
**Pull request by nikhil-varma**
Adds documentation for the `react-searchbox` library

### CODE REVIEW

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 1]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+-   **SearchComponent**: The essential headless component for implement and integrating custom UI blocks seamlessly
+
+## SearchBase
+
+This is used as an internal dependency to connect with an Elastic Search system to perform search queries. This allows adding details related to the Elastic Search system like `index`, `crendentials` etc.
+
+### Props
+
+-   **index** `string` [required]
     refers to an `index` if you’re using your own Elasticsearch cluster. If you're using an appbase.io hosted app, then the app name can be used.
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma `SearchBase` is not for apps, check the content in `SearchBase`.

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 2]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
 
     > Note: Multiple indexes can be connected to by specifying comma separated index names.
 
 -   **url** `string` [required]
     URL for the Elasticsearch cluster. Defaults to `https://scalr.api.appbase.io`
+
 -   **credentials** `string` [optional]
     Basic auth credentials for authentication purposes. It should be a string of the format `username:password`.
     If you are using an appbase.io app, you will find credentials under your [API credentials page](https://dashboard.appbase.io/app?view=credentials). If you are not using an appbase.io app, credentials may not be necessary - although having an open access to your Elasticsearch cluster is not recommended.
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: It should be `If you are using an appbase.io cluster`

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 3]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
-	showVoiceSearch
-/>
-```
-
-## Props
-
--   **app** `string` [required]
+This library has 3 essentials parts
+
+-   **SearchBase**: It is used as an internal dependency to connect with an Elastic Search system to perform search queries)
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma Why is there a bracket in the end?

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 4]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+-   **SearchBox**: The core of the library which helps with data access and manipulation with any UI components
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: The definition is not correct. It should be described as a search input component to perform. For ref. check the `DataSearch` docs.

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 5]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+-   **SearchComponent**: The essential headless component for implement and integrating custom UI blocks seamlessly
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: What do you mean by `headless` here?
It is a component to integrate any kind of `ES` query to your UI blocks.

File: content/docs/reactivesearch/react-searchbox/apireference.md (hunk 3 above)
Comment (bietkul) [reply content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma It is a provider component that provides the `searchbase` context to the child components.

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 6]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+
+## SearchBase
+
+This is used as an internal dependency to connect with an Elastic Search system to perform search queries. This allows adding details related to the Elastic Search system like `index`, `crendentials` etc.
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: Please mention it as a context provider. Also mention how one can access the `searchbase` instance from the context that we're exporting from lib.

It is missing an example, usage is not clear from the docs.

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 7]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+    -   **userId** String It allows you to define the user id to be used to record the appbase.io analytics. Defaults to the client's IP address.
+
+    -   **customEvents** Object It allows you to set the custom events which can be used to build your own analytics on top of appbase.io analytics. Further, these events can be used to filter the analytics stats from the appbase.io dashboard.
+
+-   **headers** `Object` [optional] set custom headers to be sent with each server request as key/value pairs.
+
+-   **transformRequest** `Function` Enables transformation of network request before
+    execution. This function will give you the request object as the param and expect an updated request in return, for execution.
+
+-   **transformResponse** `Function` Enables transformation of search network response before
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma Define example

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 8]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+
+    -   **enableQueryRules** Boolean If false, then appbase.io will not apply the query rules on the search requests. Defaults to true.
+
+    -   **userId** String It allows you to define the user id to be used to record the appbase.io analytics. Defaults to the client's IP address.
+
+    -   **customEvents** Object It allows you to set the custom events which can be used to build your own analytics on top of appbase.io analytics. Further, these events can be used to filter the analytics stats from the appbase.io dashboard.
+
+-   **headers** `Object` [optional] set custom headers to be sent with each server request as key/value pairs.
+
+-   **transformRequest** `Function` Enables transformation of network request before
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma Define example

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 9]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+        aggregations: {
+
+        }
+        took: 1
+    }
+    ```
+
+## SearchBox
+
+The core of the library which helps with data access and manipulation with any UI components
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma Update definition, please refer to the old docs.

File: content/docs/reactivesearch/react-searchbox/apireference.md [hunk 10]
```diff
@@ -24,120 +22,90 @@ Example uses of searchbox UI:
...
+
+### Props
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma Can you break props into sections?
Ref. `SearchBase` docs PR
Also, add examples for non-intuitive props, for e.g `defaultSuggestions`.

File: content/docs/reactivesearch/react-searchbox/quickstart.md [hunk 11]
```diff
@@ -11,39 +11,80 @@ sidebar: 'docs'
...
+```
+
+### Using yarn
+
+```js
 yarn add @appbaseio/react-searchbox
 ```
 
-##Basic Usage
+## Basic Usage
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/quickstart.md]: @nikhil-varma This example should also render the search results.

File: content/docs/reactivesearch/react-searchbox/quickstart.md [hunk 12]
```diff
@@ -11,39 +11,80 @@ sidebar: 'docs'
...
+					/>
+				</SearchBase>
 			</div>
 		);
 	}
 }
 ```
 
-##Demo
+## Result
```
Comment (bietkul) [inline content/docs/reactivesearch/react-searchbox/quickstart.md]: The title should be an example with a facet

**Review by bietkul — CHANGES_REQUESTED** [review]

File: content/docs/reactivesearch/react-searchbox/apireference.md (hunk 2 above)
Comment (siddharthlatest) [reply content/docs/reactivesearch/react-searchbox/apireference.md]: @nikhil-varma All app specific docs here should be removed. And url shouldn't default to `https://scalr.api.appbase.io` as well.

File: content/docs/reactivesearch/react-searchbox/quickstart.md [hunk 13]
```diff
@@ -11,39 +11,342 @@ sidebar: 'docs'
 nestedSidebar: 'react-searchbox-reactivesearch'
 ---
 
-[react-searchbox](https://github.com/appbaseio/react-searchbox) - It offers a lightweight (~30KB: Minified + Gzipped) and performance focused searchbox UI component to query and display results from your Elasticsearch app (aka index) using declarative props. It is an alternative to using the [DataSearch component](/docs/reactivesearch/v3/search/datasearch/) from ReactiveSearch.
+[react-searchbox](https://github.com/appbaseio/react-searchbox) - React SearchBox is a standalone, lightweight search library which allows you to add custom UI components and allow seamless search systems.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/quickstart.md:14]:
```suggestion
[react-searchbox](https://github.com/appbaseio/react-searchbox) - React Searchbox is a lightweight search library (30KB gzipped size) for quickly creating search experiences powered by Elasticsearch. It takes care of querying to Elasticsearch, managing search state and updating your search UI. Searchbox is built with ease of integration with any design system in mind.
```

Comment (siddharthlatest) [reply content/docs/reactivesearch/react-searchbox/quickstart.md:14]: I've edited this. See if this sounds better. We can make changes to this for better accuracy.

File: content/docs/reactivesearch/react-searchbox/searchbase.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
+---
+title: 'SearchBase API Reference'
+meta_title: 'API Reference for SearchBase'
+meta_description: 'SearchBase is a lightweight and platform agnostic search library for Elasticsearch.'
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbase.md:4]: Wrong meta description. It's a component. We went over this distinction in the last call.

File: content/docs/reactivesearch/react-searchbox/searchbase.md [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+    - searchbase
+    - elasticsearch
+    - search library
+sidebar: 'docs'
+nestedSidebar: 'react-searchbox-reactivesearch'
+---
+
+## SearchBase
+
+It is a container component that wraps all the ReactiveSearch components together. It binds the backend app (data source) with the UI view components
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbase.md:16]:
```suggestion
`SearchBase` is a container component that wraps all the Searchbox components together. It binds the backend index (data source) with the UI view components.
```

Comment (siddharthlatest) [reply content/docs/reactivesearch/react-searchbox/searchbase.md:16]: @nikhil-varma When you copy from other docs, make sure to edit for context of the current doc.

File: content/docs/reactivesearch/react-searchbox/searchbase.md [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+(elements wrapped within ReactiveBase), allowing a UI component to be reactively updated every time there is a change in the data source or in other UI components.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbase.md:17]: @nikhil-varma Same here. There is no `ReactiveBase`.

File: content/docs/reactivesearch/react-searchbox/searchbase.md [hunk 17]
```diff
@@ -0,0 +1,113 @@
...
+
+### Props
+
+#### To configure the Appbase environments
+
+-   **index** `string` [required]
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbase.md:23]: @nikhil-varma Refer to Searchbase docs for the prop defintions.

Environment should be without (s) suffix.

File: content/docs/reactivesearch/react-searchbox/searchbox.md [hunk 18]
```diff
@@ -0,0 +1,262 @@
...
+    - react-searchbox
+    - search library
+    - elasticsearch
+sidebar: 'docs'
+nestedSidebar: 'react-searchbox-reactivesearch'
+---
+
+## SearchBox
+
+SearchBox offers a lightweight and performance focused searchbox UI component to query and display results from your Elasticsearch cluster.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbox.md:15]: Use the definition from the updated Searchbase doc PR.

File: content/docs/reactivesearch/react-searchbox/searchbox.md [hunk 19]
```diff
@@ -0,0 +1,262 @@
...
+    Then there will be 2 suggestions from the same document
+    as we have the search term present in both the fields
+    specified in `dataField`.
+
+    ```
+    Warn
+    Washington
+    ```
+
+#### To configure the SearchBox API
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbox.md:54]: What is the SearchBox API?

File: content/docs/reactivesearch/react-searchbox/searchbox.md [hunk 20]
```diff
@@ -0,0 +1,262 @@
...
+
+-   **id** `string` [required] a unique identifier for the query can be referenced in the react property of other queries.
+
+-   **dataField** `dataFieldValidator`
+    database field(s) to be queried against. Accepts a String or an Array of either String or `DataField` type. The latter is useful for searching across multiple fields with field weights.<br/>
+    Think of field weights as a way to apply weighted search. To use field weights, you can define the `dataField` prop as an array of objects of `DataField` type.<br/>
+    The `DataField` type has the following shape:
+
+    > Note:
+    > This prop is optional only when `enableAppbase` prop is set to `true`.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbox.md:64]: @bietkul There shouldn't be any `enableAppbase` prop for this library.

File: content/docs/reactivesearch/react-searchbox/searchbox.md [hunk 21]
```diff
@@ -0,0 +1,262 @@
...
+        onAggregationData={(next, prev) => <>}
+    />
+    ```
+
+    > See impact of aggregationField with these example for [React](/docs/reactivesearch/v3/advanced/groupingresults/#how).
+
+-   **nestedField** `string`
+    Set the path of the `nested` type under which the `dataField` is present. Only applicable only when the field(s) specified in the `dataField` is(are) present under a [`nested` type](https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html) mapping.
+
+-   **size** `number` To set the number of results to be returned by a query.
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbox.md:87]: @nikhil-varma There are formatting differences b/w the prop whitespaces (description is on next line in Searchbase API ref v/s same line over here)

File: content/docs/reactivesearch/react-searchbox/searchbox.md [hunk 22]
```diff
@@ -0,0 +1,262 @@
...
+
+-   **onChange** `Function` is a callback function which accepts component's current **value** as a parameter. It is called when you are using the `value`
+    props and the component's value changes
+
+-   **onValueChange** `Function` is a callback function which accepts component's current **value** as a parameter. It is called every-time the component's value changes. This prop is handy in cases where you
+    want to generate a side-effect on value selection. For example: You want to show a pop-up modal with the valid discount coupon code when a user searches for a product in a SearchBox.
+
+-   **onValueSelected** `Function` A function callback which executes on selecting a value from result set
+
+*   **onError** `Function` gets triggered in case of an error while fetching results
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchbox.md:181]: Why are we using different bullet point chars? `-` above and `*` here.

File: content/docs/reactivesearch/react-searchbox/searchcomponent.md [hunk 23]
```diff
@@ -0,0 +1,339 @@
+---
+title: 'SearchComponent API Reference'
+meta_title: 'Documentation for SearchComponent'
+meta_description: 'React SearchBox is a lightweight react search library with some common utilities.'
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchcomponent.md:4]: @nikhil-varma Meta description should be specific to the page.

File: content/docs/reactivesearch/react-searchbox/searchcomponent.md [hunk 24]
```diff
@@ -0,0 +1,339 @@
...
+    - elasticsearch
+sidebar: 'docs'
+nestedSidebar: 'react-searchbox-reactivesearch'
+---
+
+## SearchComponent
+
+### Props
+
+#### To configure the Appbase environments
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchcomponent.md:17]: remove suffix (s)

Comment (siddharthlatest) [reply content/docs/reactivesearch/react-searchbox/searchcomponent.md:17]: Redundant article "the"

File: content/docs/reactivesearch/react-searchbox/searchcomponent.md [hunk 25]
```diff
@@ -0,0 +1,339 @@
...
+
+-   **index** `string` [required]
```
Comment (siddharthlatest) [inline content/docs/reactivesearch/react-searchbox/searchcomponent.md:19]: For all the props, refer to the issues highlighted in the SearchBox documentation review.

**Review by siddharthlatest — CHANGES_REQUESTED** [review]

**Pull request closed without merging**